File: LinkedList/swapPairs.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        temp = ListNode(0)
        temp.next = head
        cur = temp
        while cur.next != None and cur.next.next != None:
            first = cur.next
            second = cur.next.next
            cur.next = second
            first.next = second.next
            cur.next.next = first
            cur = cur.next.next; 
            #points to the second node in newly swapped pair
        return temp.next
        

File: LinkedList/swapPairs.java
class Solution {
    public ListNode swapPairs(ListNode head) {
    //1's next = 3; 0's next is 2; make 2 points to 1
                                            //head
        ListNode temp = new ListNode(0); //0-->1-->2
        temp.next = head;
        ListNode current = temp;
        while(current.next != null && current.next.next != null){
            ListNode first = current.next;
            ListNode second = current.next.next;
            current.next = second;
            first.next = second.next;
            current.next.next = first;
            current = current.next.next;
        }
        return temp.next;
    }
}